secrethub/resource_secret.go

```diff
@@ -88,8 +101,40 @@ func resourceSecretCreate(d *schema.ResourceData, m interface{}) error {
 		settings := generateList[0].(map[string]interface{})
 		useSymbols := settings["use_symbols"].(bool)
 		length := settings["length"].(int)
+		charsetSet := settings["charsets"].(*schema.Set)
+		charsets := charsetSet.List()
+		charset := randchar.Charset{}
+		if len(charsets) == 0 {
+			charset = randchar.Alphanumeric
+		}
+		if useSymbols {
+			charset = randchar.All
+		}
+		for _, charsetName := range charsets {
+			set, found := randchar.CharsetByName(charsetName.(string))
+			if !found {
+				return fmt.Errorf("could not find charset: %s", charsetName)
```

Comment:
I think we can be a bit more confident in this error message. It's not just that we went looking and didn't find a charset with that name, it doesn't exist (in this version).

```suggestion
				return fmt.Errorf("unknown charset: %s", charsetName)
```

Reply:
Same applies below when looking up the charset for the min map